elastic/kibana
# [VisTypePie] Fixes color migrations of pie charts

## Summary

This PR fixes some bugs that were identified when migrating vislib 7.13 pie charts to the new plugin. The bug has to do with applying colors (overwrite colors from legend) on some aggregations. For example on a date histogram or a range. The values on these aggs were saved differently than the es-charts. As a result they were not displayed. There was also a bug on sorting, also fixed in this PR.

### How to test it
There are two ways:
1. Create a 7.13 pie chart with a range aggregation and overwrite the colors. Migrate it to 7.14. The overwritten colors should be correct
2. You can use this SO https://gist.github.com/stratoula/918ed9433b7a140c5e00136815eb2fc6 (7.13). Check it without and with the fix. The colors should be the correct ones.

For example:
Before the fix:
![image](https://user-images.githubusercontent.com/17003240/124558500-5e3a8280-de43-11eb-9625-711f690fb2e3.png)

After the fix:
![image](https://user-images.githubusercontent.com/17003240/124558559-6abedb00-de43-11eb-9f4c-86376796cdf0.png)


Note: The bug exists on the new pie chart. If the user switches to the old implementation, the colors are depicted correctly.
